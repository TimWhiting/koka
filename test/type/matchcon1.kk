// Issue: https://github.com/koka-lang/koka/issues/9

type bar
  bar/Foo

type foo
  foo/Foo

type bar2<a>
  bar2/Foo2(a: a)

type foo2<a>
  foo2/Foo2(a: a)

fun foo/to-int(x : foo): int
  match x
    Foo -> 1

fun foo2/to-int(x: foo2<int>): int
  match x
    Foo2(y) -> y

fun main()
  foo/Foo.to-int.println
  foo2/Foo2(0).to-int.println
