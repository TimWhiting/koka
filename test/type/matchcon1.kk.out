type/matchcon1/bar/Foo: bar
type/matchcon1/bar2/Foo2: forall<a> (a : a) -> bar2<a>
type/matchcon1/bar2/a: forall<a> (bar2 : bar2<a>) -> a
type/matchcon1/foo/Foo: foo
type/matchcon1/foo/to-int: (x : foo) -> int
type/matchcon1/foo2/Foo2: forall<a> (a : a) -> foo2<a>
type/matchcon1/foo2/a: forall<a> (foo2 : foo2<a>) -> a
type/matchcon1/foo2/to-int: (x : foo2<int>) -> int
type/matchcon1/main: () -> console ()